src/msg/node.h

```diff
@@ -25,9 +25,13 @@ class CRNode {
     using job_t           = JobRunner::job_t;
     using strand_ptr_t    = std::shared_ptr<JobRunnerStrand>;
 
-    explicit CRNode() : CRNode("noname") {}
+    explicit CRNode() : CRNode("noname", nullptr) {}
 
-    explicit CRNode(std::string name) : name_(std::move(name)) {}
+    explicit CRNode(std::string name) : CRNode(std::move(name), nullptr) {}
```

Comment:
Why copying a string here? With const & you can leave it to the last one to copy.

Reply:
If a rv passed in, we can save a copy here, while the cost is only an extra move when passing in lv.

```
chenhao@Hao-Dell-Laptop ~> cat 1.cc && clang++-14 1.cc -std=c++20 -O3 -o 1 && ./1                                                                                                  [6/8130]#include <iostream>

using namespace std;

struct Foo {
    Foo() = default;
    Foo(const Foo&) { cout << "copy" << endl; }
    Foo(Foo&&) { cout << "move" << endl; }
};

void f_by_value(Foo f) {
    Foo f1(std::move(f));
}


void f_by_cref(const Foo& f) {
    Foo f1(f);
}

int main() {
    cout << "passing lv" << endl;
    Foo f;
    cout << "\nby_value" << endl;
    f_by_value(f);
    cout << "\nby_cref" << endl;
    f_by_cref(f);

    cout << "\n\npassing rv" << endl;
    cout << "\nby_value" << endl;
    f_by_value(Foo());
    cout << "\nby_cref" << endl;
    f_by_cref(Foo());
    return 0;
}
passing lv

by_value
copy
move

by_cref
copy


passing rv

by_value
move

by_cref
copy
```